compaction.go

```diff
@@ -841,6 +860,13 @@ func (d *DB) compactDiskTables(c *compaction) (ve *versionEdit, pendingOutputs [
 		meta.smallest = writerMeta.Smallest(d.cmp)
 		meta.largest = writerMeta.Largest(d.cmp)
 
+		// Bound the largest key of the sstable by the largest key of the
+		// compaction inputs. See the comment on largestInputKey for why this is
+		// necessary.
+		if db.InternalCompare(d.cmp, meta.largest, largestInputKey) > 0 {
```

Comment:
What allows `meta.largest` to exceed `largestInputKey` in the first place? Is it always a range tombstone?

Reply:
Maybe the comment on `largestInputKey` could elucidate this more and explain that naively we'd end up with the tombstone which reaches outside of the sstable creating a wider sstable as a result, but we don't want that.